src/js/gutenberg-plugin.js

```diff
@@ -184,6 +187,217 @@ const ClassifAIGenerateTagsButton = () => {
+ *
+ * @param {Object} props Props object.
+ */
+const ClassifAITSpeechSynthesisToggle = ( props ) => {
+	// State of the audio being previewed in PluginDocumentSettingPanel.
+	const [ isPreviewing, setIsPreviewing ] = useState( false );
+
+	const [ timestamp, setTimestamp ] = useState( new Date().getTime() );
+
+	// Indicates whether speech synthesis is supported for the current post.
+	let isFeatureSupported = false;
+
+	// Indicates whether speech synthesis is enabled for the current post.
+	const isSynthesizeSpeech = 'yes' === useSelect( ( select ) => select( 'core/editor' ).getEditedPostAttribute( 'classifai_synthesize_speech' ) );
+
+	// Post type of the current post.
+	const postType = useSelect( ( select ) => select( 'core/editor' ).getCurrentPostType() );
+
+	// Says whether speech synthesis is in progress.
+	const isProcessingAudio = useSelect( ( select ) => select( postAudioStore ).getIsProcessing() );
+
+	// Figure out if speech synthesis is supported by the current post.
+	if ( classifaiTextToSpeechData && classifaiTextToSpeechData.supportedPostTypes.includes( postType ) ) {
+		isFeatureSupported = true;
+	}
+
+	// The audio ID saved in the DB for the current post.
+	const defaultAudioId = useSelect( ( select ) => select( 'core/editor' ).getEditedPostAttribute( 'classifai_post_audio_id' ) );
+
+	// New audio ID in case it is regenerated manually or through publishing/updating the current post.
+	const audioId = props.audioId || defaultAudioId;
+
+	// Get the attachment data by audio ID.
+	const attachments = useSelect( ( select ) => select( 'core' ).getEntityRecords( 'postType', 'attachment', { include: [ audioId ] } ) );
+
+	// Get URL for the attachment.
+	const sourceUrl = attachments && attachments.length > 0 && attachments[ 0 ].source_url;
+
+	const isProcessingAudioProgress = useRef( false );
+
+	// Handles playing/pausing post audio during preview.
+	useEffect( () => {
+		const audioEl = document.getElementById( 'classifai-audio-preview' );
+
+		if ( ! audioEl ) {
+			return;
+		}
+
+		if ( isPreviewing ) {
+			audioEl.play();
+		} else {
+			audioEl.pause();
+		}
+	}, [ isPreviewing ] );
+
+	// Generates a unique timestamp to cache bust audio file.
+	useEffect( () => {
+		if ( isProcessingAudio ) {
+			isProcessingAudioProgress.current = true;
+		}
+
+		if ( isProcessingAudioProgress.current && ! isProcessingAudio ) {
+			setTimestamp( new Date().getTime() );
+		}
+	}, [ isProcessingAudio ] );
+
+	// Fetches the latest audio file to avoid disk cache.
+	const cacheBustingUrl = `${ sourceUrl }?ver=${ timestamp }`;
+
+	let audioIcon = 'controls-play';
+
+	if ( isProcessingAudio ) {
+		audioIcon = 'format-audio';
+	} else if ( isPreviewing ) {
+		audioIcon = 'controls-pause';
+	}
+
+	return (
+		<>
+			<ToggleControl
+				label={ __( 'Generate audio for this post.', 'classifai' ) }
+				help={ isFeatureSupported
+					? __( 'ClassifAI will generate audio for the post when it is published or updated.', 'classifai' )
```

Comment:
Would be great if we didn't hardcode to using `post` here. For instance, if I activate this feature for pages only, it still says `post` in this panel which may be confusing.